internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -39,9 +52,74 @@ func NewReceiver(logger *zap.Logger, config Config, nextConsumer consumer.Metric
 }
 
 func (r *Receiver) Start(_ context.Context, host component.Host) error {
-	return nil
+	err := r.config.validate()
+	if err != nil {
+		return fmt.Errorf("config validation failed for %q: %w", r.config.Name(), err)
+	}
+
+	configCore := r.config.monitorConfig.MonitorConfigCore()
+	monitorType := configCore.Type
+	monitorName := strings.ReplaceAll(r.config.Name(), "/", "")
+	configCore.MonitorID = types.MonitorID(monitorName)
+
+	r.monitor, err = r.createMonitor(monitorType)
+	if err != nil {
+		return fmt.Errorf("failed creating monitor %q: %w", monitorType, err)
+	}
+
+	err = componenterror.ErrAlreadyStarted
+	r.startOnce.Do(func() {
+		// starts the monitor
+		err = config.CallConfigure(r.monitor, r.config.monitorConfig)
+	})
+	return err
 }
 
 func (r *Receiver) Shutdown(context.Context) error {
-	return nil
+	err := componenterror.ErrAlreadyStopped
+	if r.monitor == nil {
+		err = fmt.Errorf("smartagentreceiver's Shutdown() called before Start() or with invalid monitor state")
+	} else if shutdownable, ok := (r.monitor).(monitors.Shutdownable); !ok {
+		err = fmt.Errorf("invalid monitor state at Shutdown(): %#v", r.monitor)
+	} else {
+		r.stopOnce.Do(func() {
+			shutdownable.Shutdown()
+			err = nil
+		})
+	}
+	return err
+}
+
+func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
+	// retrieve registered MonitorFactory from agent's registration store
+	monitorFactory, ok := monitors.MonitorFactories[monitorType]
+	if !ok {
+		return nil, fmt.Errorf("unable to find MonitorFactory for %q", monitorType)
+	}
+
+	monitor := monitorFactory() // monitor is a pointer to a monitor struct
+
+	// monitorOutputValue is the monitor struct's "Output" field value obtained via reflection to be set to
+	// an Output instance.
```

Comment:
Is this the only way we can get the `Output` field? How is this done in SmartAgent codebase?
Looks like SmartAgent uses something called ActiveMonitor to wrap the monitor instance and then sets ActiveMonitor.Output field. Would this approach also work for us?

Reply:
This effectively follows the practice of the ActiveMonitor, which informed its implementation.  We aren't incorporating ActiveMonitors directly because they're largely not exported and used as the mechanism to maintain monitor manager, which we're bypassing.

Reply:
https://github.com/signalfx/signalfx-agent/blob/4dd6371cc380f30012bc321519f02b86fd5b3e64/pkg/monitors/activemonitor.go#L121